canonical-web-and-design/maas-ui
# feat(ui): add nodedevice slice and selectors

## Done

- Added `nodedevice` slice, selectors and types
- Fetch node devices in PCI/USB tabs

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Open Redux DevTools and check there is a new item in the store: `nodedevice`
- Navigate to the PCI or USB tab of a machine and check that actions are dispatched to fetch node devices for that machine, and they successfully get added to the store

## Fixes

Fixes canonical-web-and-design/MAAS-squad#2302
